src/polyswarm_api/resources.py

```diff
@@ -1119,3 +1119,99 @@ def __init__(self, content, api=None):
         self.source = content['source']
         self.team_account_id = content['team_account_id']
         self.user_account_id = content['user_account_id']
+
+
+class ReportTask(core.BaseJsonResource):
+    RESOURCE_ENDPOINT = "/reports"
+
+    def __init__(self, content, api=None):
+        super().__init__(content, api=api)
+        self.id = content['id']
+        self.type = content['type']
+        self.format = content['format']
+        self.state = content['state']
+        self.community = content['community']
+        self.created = content['created']
+        self.template_id = content.get('template_id')
+        self.template_metadata = content.get('template_metadata', {})
+        self.sandbox_task_id = content.get('sandbox_task_id')
+        self.instance_id = content.get('instance_id')
+        self.url = content['url']
+
+    def download_report(self, folder=None):
+        """ This method is special, in that it is simply for downloading from S3 """
+        return core.PolyswarmRequest(
+            self.api,
+            {
+                'method': 'GET',
+                'url': self.url,
+                'stream': True,
+                'headers': {'Authorization': None}
+            },
+            result_parser=LocalArtifact,
+            folder=folder,
+        ).execute()
+
+
+class ReportTemplate(core.BaseJsonResource):
+    RESOURCE_ENDPOINT = "/reports/templates"
+
+    def __init__(self, content, api=None):
+        super().__init__(content, api=api)
+        if content:
+            self.id = content['id']
+            self.created = content['created']
+            self.template_name = content['template_name']
+            self.includes = content.get('includes')
+            self.excludes = content.get('excludes')
+            self.primary_color = content.get('primary_color')
+            self.footer_text = content.get('footer_text')
+            self.last_page_text = content.get('last_page_text')
+            self.is_default = content.get('is_default', False)
+            self.logo_content_length = content.get('logo_content_length')
+            self.logo_url = core.concatenate_api_uri(self.api.uri, content.get('logo_url'))
```

Comment:
I believe the server should return `template_number` somehow and we should build the logo url directly here in the resource, as show below:

```
self.logo_url = f"{self.api.uri}/v3/reports/templates/logo?id={template_number}"
```

It would get rid of the `concatenate_api_uri()`, which I think introduces a lot of complexity to the code that is not really obvious and it makes things much simpler to understand here, since we see the full url that is being built.

Reply:
Ok, I'll remove it.

Reply:
I am fine with keeping the `logo_url`, but I think a simple string concatenation here would do instead of the `concatenate_api_uri()` function.

Reply:
Removed.